api-report/sequence.api.md

```diff
@@ -78,10 +78,10 @@ export function createIdIntervalIndex<TInterval extends ISerializableInterval>()
 export function createOverlappingIntervalsIndex<TInterval extends ISerializableInterval>(client: Client, helpers: IIntervalHelpers<TInterval>): IOverlappingIntervalsIndex<TInterval>;
 
 // @public (undocumented)
-export function createOverlappingSequenceIntervalsIndex(client: Client): SequenceIntervalIndexes.Overlapping;
+export function createOverlappingSequenceIntervalsIndex(sharedString: SharedString): SequenceIntervalIndexes.Overlapping;
 
 // @public (undocumented)
-export function createStartpointInRangeIndex<TInterval extends ISerializableInterval>(helpers: IIntervalHelpers<TInterval>, client: Client): IStartpointInRangeIndex<TInterval>;
+export function createStartpointInRangeIndex<TInterval extends ISerializableInterval>(client: Client, helpers: IIntervalHelpers<TInterval>): IStartpointInRangeIndex<TInterval>;
```

Comment:
i don't know if IIntervalHelpers<TInterval> is a valid interface for consumer to know how to get or construct, my hunch is that it is not, and we should be constructing this internally.

Reply:
its also not clear to me which interval helpers get passing here, i assume they need to be for a specific interval collection or something?

the goal should be that a use can instantiate and index without needing to know anymore concepts than they do to already. if i think about that a user does, they have a shared string, they pass a label to create a collection, and then create intervals on that collection. so here, i think we should only use things they have used before like the shared string, the label, the interval colelction, or the interval. internally we should be able to find whatever we need and this must be true, as it is what we are expecting the user to do.

Reply:
If the interval type is explicitly known, such as when we have identified that an interval index is designed for `sequenceInterval` or a plain-number Interval, we can omit helper functions, as we did in the case of `createOverlappingSequenceInterval`, where we only use SharedString as the parameter without the helper. However, when the target interval types are not clear, like `startpointInRangeIndex`, we have to apply the generic type `TInterval`, this is one of the reasons we do not remove the helper function in this scenario.

Reply:
but where do helper functions come from? how does an external consumer of this API get them to pass them to the index? Can we make it so they don't have to, and we get thing within the helper function?